nixos/modules/services/web-servers/garage.nix

```diff
@@ -6,6 +6,16 @@ let
   cfg = config.services.garage;
   toml = pkgs.formats.toml { };
   configFile = toml.generate "garage.toml" cfg.settings;
+  wrapper = pkgs.writeScriptBin "garage" ''
+    # make it so all future variables set are automatically exported as environment variables
+    set -a
+
+    # source the set environmentFile (since systemd EnvironmentFile is supposed to be a minor subset of posix sh parsing) (with shell arg escaping to avoid quoting issues)
+    . ${lib.escapeShellArg cfg.environmentFile}
+
+    # exec the program with quoted args (also with shell arg escaping for the program path to avoid quoting issues there)
+    exec ${lib.escapeShellArg (lib.getExe cfg.package)} "$@"
```

Comment:
question: in which situations the path to the package can require shell quoting?

Reply:
If the `meta.mainProgram` has spaces or other sussy characters in it, it needs escaping